src/slicing/solver.rs

```diff
@@ -469,8 +475,13 @@ fn slice_sat_binary_search<'ctx>(
         prover.push();
 
         let ctx = prover.get_context();
-        let at_most_n_true = Bool::pb_le(ctx, &slice_vars, at_most_n as i32);
-        prover.add_assumption(&at_most_n_true);
+        if !slice_vars.is_empty() {
+            let at_most_n_true = match prover.get_smt_solver() {
+                SolverType::CVC5 | SolverType::YICES => at_most_k(ctx, at_most_n as i64, active_slice_vars),
```

Comment:
I'd have the function `at_most_k` take a `SolverType` parameter, and have it do the decision of whether to call `Bool::pb_le` or your own `at_most_k` implementation.